GeneralsMD/Code/Tools/WorldBuilder/src/mapobjectprops.cpp

```diff
@@ -1746,1043 +1746,1043 @@ void MapObjectProps::InitSound(void)
+	if ( loopCountEdit == NULL )
+		return;
+	
+	// If the customized checkbox is off, all customization controls are disabled
+	// If the looping checkbox is off, the loop count control is disabled
+	CButton * customizeCheckbox = ( CButton * )GetDlgItem(IDC_CUSTOMIZE_CHECKBOX);
+	CButton * loopingCheckbox = ( CButton * )GetDlgItem(IDC_LOOPING_CHECKBOX);
+	if ( customizeCheckbox == NULL || customizeCheckbox->GetCheck() == 0 ||
+			 loopingCheckbox == NULL || loopingCheckbox->GetCheck() == 0 )
+	{
+		loopCountEdit->EnableWindow( false );
+	}
+	else
+	{
+		loopCountEdit->EnableWindow( true );
+		
+		// If customization is enabled, look to see if the loop count has already been
+		// customized
+		Bool exists = false;
+		Int loopCount = 0;
+		if (m_dictToEdit) 
+		{
+			loopCount = m_dictToEdit->getInt(TheKey_objectSoundAmbientLoopCount, &exists);
+		}
+		
+		if ( exists )
+		{
+			CString loopCountText;
+			loopCountText.Format( "%d", loopCount );
+			loopCountEdit->SetWindowText( loopCountText );
+			
+			// Don't use defaults, just return
+			return; 
+		}
+	}
+	
+	// If we get here, we need the default for the loop count
+	
+	CComboBox * soundComboBox = (CComboBox *)GetDlgItem(IDC_SOUND_COMBO);
+	if ( soundComboBox == NULL )
+	{
+		loopCountEdit->SetWindowText( "0" );
+		return;
+	}
 
-  Int index = soundComboBox->GetCurSel();
-  
-  CString currentString;
-  soundComboBox->GetLBText( index, currentString );
-  if ( currentString == NO_SOUND_STRING || ( index == m_defaultEntryIndex && m_defaultIsNone ) )
-  {
-    loopCountEdit->SetWindowText( "0" );
-    return;
-  }
-  
-  if ( index == m_defaultEntryIndex )
-  {
-    // Correct the current string e.g. remove "Default <" and ">"
-    currentString = m_defaultEntryName.str();
-  }
-  
-  AudioEventInfo * audioEventInfo = TheAudio->findAudioEventInfo(static_cast< const char * >( currentString ) );
+	Int index = soundComboBox->GetCurSel();
+	
+	CString currentString;
+	soundComboBox->GetLBText( index, currentString );
+	if ( currentString == NO_SOUND_STRING || ( index == m_defaultEntryIndex && m_defaultIsNone ) )
+	{
+		loopCountEdit->SetWindowText( "0" );
+		return;
+	}
+	
+	if ( index == m_defaultEntryIndex )
+	{
+		// Correct the current string e.g. remove "Default <" and ">"
+		currentString = m_defaultEntryName.str();
+	}
+	
+	AudioEventInfo * audioEventInfo = TheAudio->findAudioEventInfo(static_cast< const char * >( currentString ) );
 
-  if ( audioEventInfo == NULL )
-  {
-    loopCountEdit->SetWindowText( "0" );
-    return;
-  }
-  
-  CString loopCountText;
-  loopCountText.Format( "%d", audioEventInfo->m_loopCount );
-  loopCountEdit->SetWindowText( loopCountText );
+	if ( audioEventInfo == NULL )
+	{
+		loopCountEdit->SetWindowText( "0" );
+		return;
+	}
+	
+	CString loopCountText;
+	loopCountText.Format( "%d", audioEventInfo->m_loopCount );
+	loopCountEdit->SetWindowText( loopCountText );
 }
 
 /// Move data from object to dialog controls
 void MapObjectProps::dictToEnabled()
```

Comment:
The github diff in this function looks suspiciously broken, which indicates that something is not right.